tools/generator/templates/player/java/interface.cc.jinja2

```diff
@@ -0,0 +1,345 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2005-2020 Association Prologin <association@prologin.org>
+{% import 'macros/cxx.jinja2' as cxx %}
+
+// This file contains the code to call the API functions from the Java language.
+// {{ stechec2_generated }}
+
+#include <cstdlib>
+#include <string>
+#include <vector>
+
+#include <jni.h>
+
+struct Champion
+{
+    static jclass Class();
+};
+{% for enum in game.enum %}
+
+{{ cxx.decl_enum(enum) }}
+
+struct {{ enum.enum_name|camel_case }}
+{
+    static jclass Class();
+};
+{% endfor %}
+{% for struct in game.struct %}
+
+{{ cxx.decl_struct(struct) }}
+
+struct {{ struct.str_name|camel_case }}
+{
+    static jclass Class();
+};
+{% endfor %}
+
+struct StechecJavaRunTime
+{
+    StechecJavaRunTime() = default;
+    ~StechecJavaRunTime();
+    void init();
+    bool function_enter();
+    void function_exit(bool attached);
+
+    JavaVM* jvm;
+    JNIEnv* env;
+    jobject prologin;
+
+private:
+    bool is_init_ = false;
+};
+static struct StechecJavaRunTime jrt;
+
+{% for func in game.function %}
+extern "C" {{ func|cxx_prototype(prefix='api_') }};
+{% endfor %}
+{% for enum in game.enum %}
+extern "C" void api_afficher_{{ enum.enum_name }}({{ enum.enum_name }} v);
+{% endfor %}
+{% for struct in game.struct %}
+extern "C" void api_afficher_{{ struct.str_name }}({{ struct.str_name }} v);
+{% endfor %}
+
+// In case of errors...
+template <typename Lang, typename CxxType>
+CxxType java_to_cxx(Lang in)
+{
+    return in.error_should_not_happen;
+}
+
+template <typename CxxType, typename Lang>
+Lang cxx_to_java(CxxType in)
+{
+    return in.error_should_not_happen;
+}
+
+template<typename Lang, typename CxxType>
+std::vector<CxxType> java_to_cxx_array(jobject in);
+
+template<typename CxxType, typename Lang>
+jarray cxx_to_java_array(std::vector<CxxType> in);
+
+// Basic type wrappers
+{% for type in ['bool', 'int', 'double'] %}
+{% set arraytype = type|jni_type + 'Array' %}
+template <>
+{{ type|jni_type }} cxx_to_java<{{ type }}, {{ type|jni_type }}>({{ type }} in)
+{
+    return ({{ type|jni_type }})in;
```

Comment:
I think we put spaces after casts in our format options

Reply:
I ran clang-format to see and we don’t put spaces.